### DISCUSSION TITLE
Best way to set up automatic backup logic (backup Android phone folders, Windows computer folders, etc)

### DISCUSSION THREAD

**Discussion by logless** [drakkan/sftpgo#1652] [General]
I quickly got sftpgo web interface up and running over HTTPS with my own domain. However, my main goal is to set up crons for backing up certain folders from a Windows computer (using Task Scheduler) and certain folders from an Android phone. 

I understand there are probably many ways to do that, but which one would be the most straightforward and lightweight?

I guess for Android I need an app that supports scheduled tasks over SSH/FTP/WEBDAV. Any recommendations?

For Windows, a simple CLI tool that can log in to the SFTPGO server and upload some folders would be sufficient. 

Thanks in advance for the help.

**Comment by logless** [comment]
Alright, I somehow managed to find a simple way to do it, so I add it here for reference. There are quite a lot of people looking to set up a personal cloud server, but the most popular solutions - NextCloud and FileCloud are incredibly heavyweight and would require expensive cloud hardware to run properly, which often makes them unfeasible. 

So, an easy way to sync folders on Windows would be using RCLONE - a simple .exe that can be added to PATH for more convenient usage and then run through Task Scheduler in the form of a .bat file (in Task Scheduler, the actual program to run should be cmd.exe with the batch file as a command line parameter (cmd.exe /C F:\path\to\script.bat)

to set up RCLONE, run `rclone config` in order to create the configuration file, inputting the login details of a user you have created in SftpGo admin interface and your server address and port. You will set a name for the remote server as well. 

Then, just run rclone sync C:\Mydocuments YOUR_SFTPGO_SERVER_NAME:/remote_path/some_dir to sync a local directory to the cloud. 

On Android, it gets a bit funnier. On Samsung Galaxy phones you can set up your SftpGo server as a Network Storage in My Files, which is great and works perfect. However, I have not found a way to automate backups using it. 

So, what I did was:

1) Install F-Droid
2) Install Termux
3) Make sure the Termux app have Files and Folders permission
4) In Termux console, run `pkg install rclone cronie termux-services `
5) then run `sv-enable crond`
5) `crontab -e` to open the crontab and add your entry
Example:
`* */2 * * * rclone sync /storage/emulated/0/DCIM/Camera YOUR_SFTPGO_SERVER_NAME:/remote_path/some_dir`

6) Restart Termux

Note: /storage/emulated/0/ is the path to your phone storage, that's what you usually see when you connect it over USB to a computer.
I still haven't tested if Termux runs the cron regularly when the app is completely closed. There are at least ways to make the app run automatically after phone restart. 

But anyway, ideally the in-built Network Storage functionality of Samsung phones would be used for syncing, but I haven't found a way to do that yet.

**Reply by thielj** [reply]
Try Kopia, it works on Linux/Win/Mac
